route/src/route_generator_worker.cpp

```diff
@@ -438,5 +438,98 @@ namespace route {
         return true;
     }
 
+bool RouteGeneratorWorker::crosstrack_error_check(const geometry_msgs::PoseStampedConstPtr& msg, lanelet::ConstLanelet current, carma_wm::TrackPos llt_track)
+{
```

Comment:
At the very beginning, if the point is within the lanelet, we will be optimizing it as we don't need to check run rest of the function. So including `boost::geometry::within(position, current.polygon2d())` if logic here might be a good idea

Reply:
@lewisid pls add the protection here.